core/internal/stream/sender.go

```diff
@@ -1573,8 +1583,18 @@ func (s *Sender) sendExit(record *spb.Record) {
 }
 
 // sendMetric updates the metrics in the run config.
-func (s *Sender) sendMetric(metric *spb.MetricRecord) {
-	err := s.runConfigMetrics.ProcessRecord(metric)
+func (s *Sender) sendMetric(record *spb.Record, _ *spb.MetricRecord) {
+	// if server side expand glob metrics is enabled, we don't need to send internal metrics
+	// as these were expanded internally by the client.
```

Comment:
Don't we still need to upload defined metrics in the run config?

And what would happen on resume?

Reply:
i will create a follow up ticket, to handle define_metric correctly in the resume case and include the complication of server/client expansion